ui/ui-components/client/client.ts

```diff
@@ -107,33 +112,39 @@ export class Client {
     } else if (code === "AUTHORIZATION_ERROR") {
       // ok, it will be rendered on the page
     }
-  }
+  };
 
-  csrfToken() {
+  private csrfToken = () => {
     if (globalThis?.localStorage) {
       return window.localStorage.getItem("session:csrfToken");
     }
-  }
+  };
+
+  private static readonly optionDefaults = {
+    useCache: true,
+    errorHandler,
+    uploadHandler,
+  };
 
-  async action<Response = any>(
+  public request = async <Response = any>(
     verb: Method = "get",
     path: string,
     data: AxiosRequestConfig["data"] = {},
-    useCache = true,
-    uploadHandler?: UploadHandler,
-    req?: IncomingMessage,
-    res?: ServerResponse
-  ): Promise<Response> {
+    options: Partial<typeof Client.optionDefaults> = {}
```

Comment:
Maybe the Options should be explicit instead of derived from `optionDefaults`. There is no optionDefaults can be broken at any time without having the right Options contract

Reply:
I'm not sure what you're getting at here. optionDefaults are defaulted, and you can't pass anything not in it. They're override only.

Reply:
My point is that it's strange that this is deriving which options are valid from a constant that assigns default values Using explicit typing is a lot stronger way to establish what's valid.

Reply:
Fair enough, added an interface.